src/util-inl.h

```diff
@@ -562,6 +562,22 @@ constexpr std::string_view FastStringKey::as_string_view() const {
   return name_;
 }
 
+// Converts a V8 numeric value to a corresponding C++ primitive or enum type.
+template <typename T>
```

Comment:
I think you can follow the numeric version of ToV8Value and use a SFINAE with `std::enable_if<std::numeric_limits<T>::is_specialized, bool>::type>`.